### ISSUE TITLE
`parseMLST` returns the wrong filename if it ends with a space

### ISSUE THREAD

**Issue by Quezler** [nullobsi/ftpdeno#8]
First of all, thanks for fixing #7 as well as all the other commits released at that time that significantly improved this codebase.

Currently gladly using `extendedList` instead of a `list` + `stat` call for each, even with caching that would have been bad.

===

Now for the issue: if a filename ends on a space (which just so happened to be the case on the ftp server i am testing against) the parser trims the `;` separated list, and the filename just so happens to be the last in that list in my case.

I have since removed the trailing space from that one directory name, but not everyone might have that same luxury.

I have narrowed the issue down to this line:
https://github.com/nullobsi/ftpdeno/blob/85e273a4957e1beb2a85c928938ee3fa86c3dc1c/classes/FTPClient.ts#L510

As well as the example data, edited in a fork since that would be easy to import into my current deno project:
https://github.com/Quezler/ftpdeno/blob/3a646de6f0040c6cc2c3657214b1d07d104038e3/classes/FTPClient.ts#L512-L514

```
"modify=20221025221143;perm=flcdmpe;type=dir;unique=801U1420148;UNIX.group=1006;UNIX.groupname=psacln;UNIX.mode=0755;UNIX.owner=10032;UNIX.ownername=[redacted]; 25-10-2022 "
[
    "modify=20221025221143",
    "perm=flcdmpe",
    "type=dir",
    "unique=801U1420148",
    "UNIX.group=1006",
    "UNIX.groupname=psacln",
    "UNIX.mode=0755",
    "UNIX.owner=10032",
    "UNIX.ownername=[redacted]",
    " 25-10-2022"
]
```

**Comment by nullobsi** [comment]
Aha. Yeah I was trying to consider if the filename ends/starts with a space, but I forgot that trim also does the end and therefore the last line.

**Comment by nullobsi** [comment]
Alright, I think I fixed this in v2.0.1 :)
